### PR TITLE
get and delete user policy tests

### PR DESCRIPTION
**Pull request by ravindrachoudhari**
Adding Get user policy and Delete user policy tests for [F-20E: IAM policy support](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/968458960/F-20E+IAM+Policy+Support)

Test Design doc: https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/980255029/F-20E+S3+API+IAM+Policy+Support+-+PI7

### CODE REVIEW

File: s3tests_boto3/functional/test_iam.py [hunk 1]
```diff
@@ -0,0 +1,440 @@
+import json
+from operator import eq
+
+from botocore.exceptions import ClientError
+from nose.plugins.attrib import attr
+
+from s3tests.functional.utils import assert_raises
+from . import get_tenant_iam_client, get_tenant_user_id
```
Comment (apoorva-rao) [inline s3tests_boto3/functional/test_iam.py]: Let's try and follow the import conventions followed in s3tests_boto3/functional/test_s3.py

**Review by apoorva-rao — APPROVED** [review]
@ravindrachoudhari added one minor comment to improve consistency across the files. 

If possible let's also add a couple of lines to the commit message to describe the purpose of this PR.

Approving to unblock the PR.

File: s3tests_boto3/functional/test_iam.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+    client = get_tenant_iam_client()
+
+    policy_document = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": "*",
+             "Resource": "*"}}
+    )
+    client.put_user_policy(PolicyDocument=policy_document, PolicyName='AllAccessPolicy',
```
Comment (ketanarlulkar) [inline s3tests_boto3/functional/test_iam.py]: Deleting the applied policy as part of cleanup, is missing.

**Review by ketanarlulkar — APPROVED** [review]

File: s3tests_boto3/functional/test_iam.py [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+                           UserName=get_tenant_user_id())
+
+    e = assert_raises(ClientError, client.get_user_policy, PolicyName='AllAccessPolicy',
+                      UserName="some-non-existing-user-id")
+    status = _get_status(e.response)
```
Comment (seagate-sarang-sawant) [inline s3tests_boto3/functional/test_iam.py:241]: Are you planning to validate error response messages?

**Review by seagate-sarang-sawant — APPROVED** [review]

**Pull request merged**